### DISCUSSION TITLE
Request for making NWD Themes Revolution Slider 6.1.1.6 compatible with OpenMage 20.1.0

### DISCUSSION THREAD

**Discussion by matteotestoni** [OpenMage/magento-lts#3575] [Hire]
https://github.com/OpenMage/magento-lts/issues/3572

NWDthemes Revolution Slider - https://nwdthemes.com/extensions/slider-revolution/

upload image to a slide
save slide

openmage logout as soon you click "upload"
in prototype.js parsejson redirect to login page and openmage is now logged out from admin
Maybe cookie change in openmage?
Extension also search for fusty-flow.js, fusty-flow-factory.js ext-all.css removed from openmage 20.1 rc2

**Comment by pquerner** [comment]
IE Support was removed, yes. But I dont understand the other issues you are having. While you logout you clear cookies, and some ajax call fails (ie. the saving/uploading, with now defunct old cookies). Thats expected, no?

// edit
just now realized this was coming from this report: https://github.com/OpenMage/magento-lts/issues/3572

with more context perhaps

**Comment by kiatng** [comment]
Have you try the vendor's support: https://support.nwdthemes.com/login  Would they support OpenMage? If not, would they provide a test copy for OpenMage maintainers for testing purpose only?

**Reply by matteotestoni** [reply]
Of Course.

_Hi Matteo
Unfortunately we stopped releasing the version for Magento 1 some time ago. 
The last one released for Magento 1 was 6.1.1.6 and was compatible with Magento 1.9.4.5.
We have not tested Magento Slider Revolution with OpenMage but as the Openmage 20.x is based on Magento 1.9.4.5 - it should be compatible with it. Or incompatibility issues could be resolved if anything is found.
Cheers @ NWDthemes_

And me

_Hi,
there are many website that use openmage
https://trends.builtwith.com/shop/OpenMage
Is possible a release of your extension compatible with openmage 20.1?_

**Reply by matteotestoni** [reply]
_Hi Matteo
Yes, we have thoughts about resuming the release of the Slider Revolution module for OpenMage some time ago but still in doubt if it will be worth it.
Btw are you maintainer of the OpenMage project, right?
Sure we can provide you with the latest source of Slider Revolution for Magento 1.9.4.5 so you can test it with OpenMage.
And if it is possible to give us a favour as well by putting a link to us or our product or maybe some article?_

**Reply by matteotestoni** [reply]
[kiatng](https://github.com/kiatng) maybe can you email NWD extension and maybe can be ported to openmage 20.1

**Reply by kiatng** [reply]
Sorry, I don't have the extension.

**Comment by matteotestoni** [comment]
https://vimeo.com/870963768
At the end i refresh page but session is not valid and OpenMage redirect to login page

**Comment by pquerner** [comment]
And what was the URL for that ajax call and the actual response?

**Comment by matteotestoni** [comment]
https://www.bikedirection.com/index.php/admin/nwdimages/upload/type/image/key/136f17c28dfa0ba8bf953c6bd6382822/?SID=6p1d6tvblqjodhguj83fgt0n3k

**Reply by pquerner** [reply]
When you were in the dev console and saw the `unexpected token` error, what was the response from the backend? Maybe it has some error text attached to it to help identify the problem.

**Comment by matteotestoni** [comment]
response is html of login page so is not a json and there is error. After that if i refresh page openmage go to login
There is something in upload that delete login cookie or something similar

**Reply by pquerner** [reply]
Ok. Do you see anything in the magento error log or php's error log?

**Reply by matteotestoni** [reply]
nothing. i've tried adding additional log but nothing.
NWD extend magento 1.9.4.5 uploader so maybe there are incompatibilities

**Reply by matteotestoni** [reply]
i have zip of extension. Can you work on it and how much can be your price? more or less?

**Reply by pquerner** [reply]
Can you create a local instance of OpenMage and not update to 20.1 right away? Try the latest 19.4.x, with IE support in-tact. It might be that the uploader acutually requires the old IE files - for whatever reason.
Pretty much anything here is just poking blindly. Without actually having access to the sourcecode theres little one can do.

Or if you want to stay on 20.1.x - but do that locally - add Mage::log calls (or better yet Xdebug) to the NWD Uploadcontroller file and find out where it breaks. Maybe it also just helps enableding Mage Developer Mode and something isnt compatible with the PHP version you are running.

**Reply by matteotestoni** [reply]
OpenMage 19.4.x Works
i need openmage 20.1 for ga4 module
I'll try add mage::log to controller. 

I have a beta site with ftp access if you want

**Reply by pquerner** [reply]
If you want send me the .zip of the module to my email at openmage@querner.dev. I will take a look.

**Reply by matteotestoni** [reply]
zip sended

**Reply by pquerner** [reply]
This module needs Mage_Backup, which was removed from the core and can be found in a seperate repository now:
https://github.com/OpenMage/module-mage-backup

But sorry I found no easy way to fix the behaviour. Maybe someone else can take a look.
For me I couldnt even find that the correct controller was called. In the preDispatch I was logged out as soon as I came in there (after clicking the upload button) - so most likely something in between, maybe defunct javascript that plays with cookies? 

\Nwdthemes_Revslider_Adminhtml_NwdimagesController::uploadAction was never called for me.

**Reply by sreichel** [reply]
> For me I couldnt even find that the correct controller was called.

Did we have a PR that removed old admin routing?

**Reply by fballiano** [reply]
https://github.com/OpenMage/magento-lts/pull/1551 is in next

**Comment by pquerner** [comment]
> > For me I couldnt even find that the correct controller was called.
> 
> Did we have a PR that removed old admin routing?

I mean the controller was called. I verified with preDispatch and indexAction. Just the other actions (specially uploadAction) did not trigger.

Clicking the "Upload files" button came to (a) controller (only checked preDispatch) and the adminsession was already logged out (atleast reported like so).

**Reply by sreichel** [reply]
Mhh, w/o code ..... pls forward zip if its okay.

**Reply by matteotestoni** [reply]
Which email?

**Reply by sreichel** [reply]
github-sr@hotmail.com

@ADDISON74 this is good example why I'm not okay with changed release strategy.

Users are forced to use v20 to get improvements, but run into b/c breaks. Burring v19 was a big mistake imho.

**Reply by addison74** [reply]
@sreichel - Please don't lose sight of an important aspect, time passes, new versions of PHP, MySQL and so on are released. All Magento v1 or OpenMage 1.9 users should be aware of the changes and consider updating their base platform, then custom work like extensions, modules. I believe that everyone still has a long period of time to update what they use. However, they were not asked to make an immediate big step, they can still choose to use v19 or switch to v20. This case related to a slider is related to v20, but such a step cannot be taken without detailed investigations.

**Reply by matteotestoni** [reply]
You can keep a database of extensions to keep them updated by asking permission from the developers.
For example NWD is available, I forwarded the email to you but we need someone to do PR

**Reply by pquerner** [reply]
@matteotestoni I've shared the module privately to @sreichel and @kiatng , hope you don't mind.

**Reply by matteotestoni** [reply]
NWD write to me: 
_Hi Matteo
Yes, we have thoughts about resuming the release of the Slider Revolution module for OpenMage some time ago but still in doubt if it will be worth it.
Btw are you maintainer of the OpenMage project, right?
Sure we can provide you with the latest source of Slider Revolution for Magento 1.9.4.5 so you can test it with OpenMage.
And if it is possible to give us a favour as well by putting a link to us or our product or maybe some article?_

**Reply by sreichel** [reply]
> Please don't lose sight of an important aspect

Therefor we had that v20 branch - that was only for bc breaking changes ... everthing else went to v19. Every fix, every feature. To switch from Magento to OpenMage was a sure shot for years. But now ... maybe its better to suggest M2.  :(

When i need to fix my modules to work with OM, i can also convert them for M2 in minutes ... 

Instead of improving code quality (continue work on phpstan-issues, add templates to sniffs, ...) new features are planed/introduceded. 

Why do we have a "next" branch when current one already allows breaking changes?

**Reply by Flyingmana** [reply]
please move the Versioning and BC topic into a separate discussion or a new rfc

**Reply by addison74** [reply]
I think you agree with me that as many custom changes exist for Magento/OpenMage v19, there are obviously BCs that we don't know about. I doubt that you could say that v19 guaranteed does not create BCs.

If you want to discuss this topic, please open another thread. Someone wants to solve a problem and we start talking over him.

**Reply by sreichel** [reply]
Last reply ... of course I can guarantee that old v19 had no breaking changes. PRs have been checked to not break anything.

**Reply by addison74** [reply]
If you still write a last post, I appreciate your opinion but I am not trying to make big efforts to change it. I have a store now in the works that has Magento 1.9.2.4 running many custom extensions and code on a Debian 10 with PHP 7.3 that will be updated to a Debian 12 with PHP 8.2. Initially I tested OpenMage v20 but there are enough problems that need solving, then I wanted to check with OpenMage v19, no change for the better. So someone who has an old Magento store will make an effort to reach OpenMage v20. Does v19 have any future since an RFC was voted? No one should remain captive in v19 precisely because things are constantly changing. As long as a store is working, no one should go to it, but problems arise when you have to migrate to another hosting service provider, years have passed, PHP versions have brought changes and many others. You said that you use Windows 7, but if you use Adobe products, you will notice that you must have at least Windows 10 22H2 to be able to install Photoshop 25.

**Comment by kiatng** [comment]
I installed the extension in v20. It has over 3100 files, It depends on `Mage_Backup`, which was removed. I do not know how to replicate the error. Please provide steps.

**Reply by matteotestoni** [reply]
https://vimeo.com/870963768

**Reply by pquerner** [reply]
When it calls the Mage_Backup thingy you can add a early return so the Model isnt loaded, thus no conflict. Thats how I used it in 20.1.1.

**Comment by kiatng** [comment]
Try this:

in app\code\community\Nwdthemes\Revslider\Block\Adminhtml\Cms\Wysiwyg\Images\Content\Uploader.php


Replace line 47:
```php
Mage::getModel('adminhtml/url')->addSessionParam()->getUrl('*/*/upload', array('type' => $type))
```
with this:
```php
$this->getUrl('*/*/upload', array('type' => $type))
```

![image](https://github.com/OpenMage/magento-lts/assets/1106470/dba28869-110c-476a-b901-1bcd9a059879)

I hope the fix works for you.

**Reply by matteotestoni** [reply]
Now don't logout anymore and works but i have an error that after few seconds go away

jaguar-i-pace-1-removebg-preview.png (129.7 kB)Invalid method Nwdthemes_Revslider_Model_Images_Storage::_getSession(Array ( ) )

**Reply by matteotestoni** [reply]
thank you

**Reply by pquerner** [reply]
`_getSession` is probably the same as `getSession` as said here: https://github.com/OpenMage/magento-lts/issues/3572#issuecomment-1744794352

**Reply by kiatng** [reply]
About the missing method `_getSession()`, perhaps you can add the method 

```php
    public function _getSession()
    {
        return Mage::getSingleton('adminhtml/session');
    }
```

in app\code\community\Nwdthemes\Revslider\Model\Images\Storage.php
